src/Datadog.Trace.ClrProfiler.Managed/Integrations/AspNetScopeManager.cs

```diff
@@ -31,20 +40,36 @@ public Scope Activate(Span span, bool finishOnClose)
         {
             var activeScope = Active;
             var scope = new Scope(activeScope, span, this, finishOnClose);
+
+            var scopeDeactivatedArgs = new ScopeEventArgs(activeScope);
+            var scopeOpenedArgs = new ScopeEventArgs(scope);
+
+            ScopeOpened?.Invoke(this, scopeOpenedArgs);
+
             SetScope(scope);
+            ScopeDeactivated?.Invoke(this, scopeDeactivatedArgs);
```

Comment:
Same idea as below, should we notify ScopeDeactivated if the active scope was null?

Reply:
Same, I'll fix that 👍